File: Java/src/UI/TextUI.java
package UI;

import irrgarten.Directions;
import irrgarten.GameState;
import java.util.Scanner;


public class TextUI {
    
    private static Scanner in = new Scanner(System.in);
    
    private char readChar() {
        String s = in.nextLine();     
        return s.charAt(0);
    }
    

    public Directions nextMove() {
        System.out.print("Where? ");
        
        Directions direction = Directions.DOWN;
        boolean gotInput = false;
        
        while (!gotInput) {
            char c = readChar();
            switch(c) {
                case 'w':
                    System.out.print(" UP\n\n");
                    direction = Directions.UP;
                    gotInput = true;
                    break;
                case 's':
                    System.out.print(" DOWN\n\n");
                    direction = Directions.DOWN;
                    gotInput = true;
                    break;
                case 'd':
                    System.out.print("RIGHT\n\n");
                    direction = Directions.RIGHT;
                    gotInput = true;
                    break;
                case 'a':
                    System.out.print(" LEFT\n\n");
                    direction = Directions.LEFT;
                    gotInput = true;    
                    break;
            }
        }   
        return direction;
    }
    
    public void showGame(GameState gameState) {  
        
        System.out.println(gameState.getLabyrinth());
        System.out.println("Lista de jugadores: "+gameState.getPlayers());
        System.out.println("Lista de monstruos: "+gameState.getMonsters());
        System.out.println("Jugador actual: "+gameState.getCurrentPlayer());
        System.out.println("Ganador?: "+gameState.getWinner());
        System.out.println(gameState.getLog());
        
        
        
        
        
        
        
        
        
    }
    
}

File: Irrgarten-Ruby/UI/textUI.rb

require 'io/console'
require_relative '../irrgarten/directions'
require_relative '../irrgarten/game_state'

module UI

  class TextUI

    #https://gist.github.com/acook/4190379
    def read_char
      STDIN.echo = false
      STDIN.raw!
    
      input = STDIN.getc.chr
      if input == "\e" 
        input << STDIN.read_nonblock(3) rescue nil
        input << STDIN.read_nonblock(2) rescue nil
      end
    ensure
      STDIN.echo = true
      STDIN.cooked!
    
      return input
    end

    def next_move
      print "Where? "
      got_input = false
      while (!got_input)
        c = read_char
        case c
          when "w"
            puts "UP ARROW"
            output = Irrgarten::Directions::UP
            got_input = true
          when "s"
            puts "DOWN ARROW"
            output = Irrgarten::Directions::DOWN
            got_input = true
          when "d"
            puts "RIGHT ARROW"
            output = Irrgarten::Directions::RIGHT
            got_input = true
          when "a"
            puts "LEFT ARROW"
            output = Irrgarten::Directions::LEFT
            got_input = true
          when "W"
            puts "UP ARROW"
            output = Irrgarten::Directions::UP
            got_input = true
          when "S"
            puts "DOWN ARROW"
            output = Irrgarten::Directions::DOWN
            got_input = true
          when "D"
            puts "RIGHT ARROW"
            output = Irrgarten::Directions::RIGHT
            got_input = true
          when "A"
            puts "LEFT ARROW"
            output = Irrgarten::Directions::LEFT
            got_input = true
          when "\u0003"
            puts "CONTROL-C"
            got_input = true
            exit(1)
          else
            #Error
        end
      end
      output
    end

    def show_game(game_state)
      puts game_state.labyrinth.to_s
      puts "Lista de jugadores: #{game_state.players}"
      puts "Lista de monstruos: #{game_state.monsters}"
      puts "Jugador actual: #{game_state.current_player}"
      if game_state.winner
        puts "Hay algún ganador?: Sí, el jugador #{game_state.current_player} ha ganado la partida"
      else
        puts "Hay algún ganador?: No"
      end
      puts game_state.log.to_s
      puts "==============================================================="

    end

  end # class   

end # module   


